server/jetstream_cluster.go

```diff
@@ -3609,22 +3686,33 @@ func groupName(prefix string, peers []string, storage StorageType) string {
 
 // createGroupForStream will create a group for assignment for the stream.
 // Lock should be held.
-func (cc *jetStreamCluster) createGroupForStream(ci *ClientInfo, cfg *StreamConfig) *raftGroup {
+func (js *jetStream) createGroupForStream(ci *ClientInfo, cfg *StreamConfig) *raftGroup {
 	replicas := cfg.Replicas
 	if replicas == 0 {
 		replicas = 1
 	}
-	cluster := ci.Cluster
-	if cfg.Placement != nil && cfg.Placement.Cluster != _EMPTY_ {
+
+	// Default connected cluster from the request origin.
+	cc, cluster := js.cluster, ci.Cluster
+	// If specified, override the default.
+	clusterDefined := cfg.Placement != nil && cfg.Placement.Cluster != _EMPTY_
+	if clusterDefined {
 		cluster = cfg.Placement.Cluster
 	}
+	clusters := []string{cluster}
+	if !clusterDefined {
+		clusters = append(clusters, ci.Alternates...)
+	}
+
 	// Need to create a group here.
-	// TODO(dlc) - Can be way smarter here.
-	peers := cc.selectPeerGroup(replicas, cluster)
-	if len(peers) == 0 {
-		return nil
+	for _, cn := range clusters {
+		peers := cc.selectPeerGroup(replicas, cn, cfg)
```

Comment:
Fine for now I guess but in real world I think we'd need a lot more control here - probably once tag placement is in. 

But like even now it would be pretty bad if the next nearest cluster with available space is lets say Asia while I am connected to USA?

Reply:
We have plans on continuing to extend the resource utilization used in the decision, right now this is a step up for random placement.

Also note that the system will not overflow if you provide placement guidance, so you can opt-out and have the stream create request fail.